### DISCUSSION TITLE
[Draft] Breaking changes in Rspack v0.7.0

### DISCUSSION THREAD

**Discussion by LingyuCoder** [web-infra-dev/rspack#6218] [Core]
This thread is used to collect planned breaking changes for Rspack v0.7.0.

**Comment by ahabhgk** [comment]
remove resolve.browserField: https://github.com/web-infra-dev/rspack/blob/5f65349e2ca1f3c60ed4564fe379a72f8ca17c71/packages/rspack/src/config/zod.ts#L385-L399

**Reply by h-a-n-a** [reply]
remove `experiments.newSplitChunks`:
https://github.com/web-infra-dev/rspack/blob/5f65349e2ca1f3c60ed4564fe379a72f8ca17c71/packages/rspack/src/config/zod.ts#L1198-L1211

**Comment by ahabhgk** [comment]
remove `builtins.treeShaking` (oldTreeShaking) and `builtins`

**Comment by hardfist** [comment]
switch experiments.css to false by default @JSerFeng

**Reply by JSerFeng** [reply]
Postponed to 0.8

**Comment by hardfist** [comment]
enable concatmodulePlugin by default in production @JSerFeng

**Reply by JSerFeng** [reply]
Cancelled

**Comment by xc2** [comment]
~~https://github.com/web-infra-dev/rspack/pull/6370~~

~~Not sure if this fix is a breaking change:~~

~~envvar `process.env.WEBPACK_SERVE` on build mode will be `undefined` instead of `"true"`~~

**Reply by chenjiahan** [reply]
I prefer this to be a fix or improvement rather than a breaking change 😄

**Comment by h-a-n-a** [comment]
Bump `swc_core` to the latest minor (`0.91.x`), which includes a breaking change to ast, thus to SWC wasm plugin.

**Reply by h-a-n-a** [reply]
Note: This is not necessarily to be released in the next version. Listing here for a heads-up.

**Comment by ahabhgk** [comment]
Remove `exportsConvention` for module type `css`, [it's an mistake that introduce it for module type `css`](https://github.com/webpack/webpack/pull/18341)

**Comment by LingyuCoder** [comment]
Remove builtin define plugin in rspack-cli https://github.com/web-infra-dev/rspack/pull/6219, relate to https://github.com/web-infra-dev/rspack/issues/6214

**Comment by h-a-n-a** [comment]
Deprecating and stabilizing JavaScript APIs in https://github.com/web-infra-dev/rspack/pull/6448.
API-wise detail will be added once finished.

**Reply by h-a-n-a** [reply]
- Compiler/MultiCompiler: https://github.com/web-infra-dev/rspack/pull/6448

**Reply by h-a-n-a** [reply]
- Compilation: https://github.com/web-infra-dev/rspack/pull/6505

**Comment by JSerFeng** [comment]
Remove old treeshaking
